controlplane/pkg/plugins/utils.go

```diff
@@ -94,6 +114,41 @@ func registerPlugin(name, endpoint, registry string, capabilities []plugins.Plug
 		Endpoint:     endpoint,
 		Capabilities: capabilities,
 	})
+	if err != nil {
+		logrus.Errorf("%s: failed to register plugin: %s, grpc code: %+v", name, err, status.Convert(err).Code())
+		return err
+	}
+
+	logrus.Infof("%s: starting NSM liveness monitoring", rm.clientName)
+	rm.livenessClient, err = livemonitor.NewClient(conn)
+	if err != nil {
+		logrus.Errorf("%s: failed to get NSM liveness channel: %s, grpc code: %+v", name, err, status.Convert(err).Code())
+		return err
+	}
+
+	go rm.monitor()
 
 	return err
 }
+
+func (rm *registrationMonitor) monitor() {
+	for {
+		select {
+		case <-rm.livenessClient.Context().Done():
+			logrus.Infof("%s: finishing NSM liveness monitoring", rm.clientName)
+			if rm.onDone != nil {
```

Comment:
What `onDone` stands for? And why this is ok for us to stop monitoring if the context is done?